### PR TITLE
[Android] Allow custom background drawable with a custom handler

### LINKED ISSUE #18720: [regression/8.0.0-preview.4.8333] [Android] Setting the background property of AppCompatEditText in a handler mapping does not work
### Description

In a .NET MAUI project using the latest .NET 8 RC, setting the background property of a `AppCompatEditText` doesn't seem to work.

Using this code:

```csharp
Microsoft.Maui.Handlers.EntryHandler.Mapper.AppendToMapping("TestEntry", (handler, view) =>
{
    handler.PlatformView.Background = null;
    handler.PlatformView.SetBackgroundColor(Android.Graphics.Color.Pink);
    handler.PlatformView.SetTextColor(Android.Graphics.Color.Aqua);
});
```

Has the following results:

image

**❌ Figure: PlatformView background properties not applied, but other mappings are applied**

image

**✅ Figure: The same code in .NET 7**

Repro is linked below. Repo with working version in .NET 7: https://github.com/matt-goldman/Net7AndroidHandlers

### Steps to Reproduce

1. Create a .NET MAUI app in .NET 8
2. Create a handler mapping to set the background of the `PlatformView` for `Entry`
3. Add an `Entry` to your UI
4. Run the app, observe bug 🐞

### Link to public reproduction project repository

https://github.com/matt-goldman/Net8AndroidHandlers

### Version with bug

8.0.0-preview.4.8333

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.0-preview.3.8149

### Affected platforms

Android

### Affected platform versions

Android - all versions

### Did you find any workaround?

I could not find a workaround

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Allow custom background drawable with a custom handler on Android.

| Entry | Editor | Time Picker | Date Picker |
|:-:|:-:|:-:|:-:|
| fix-18720-entry | fix-18720-editor | fix-18720-timepicker | fix-18720-datepicker |

Note, as in the previous gif, we can add/remove the `BackgroundColor `or `Background `in addition to being able to customize the control with a custom handler, removing the background or setting a custom drawable.

### Issues Fixed

Fixes #18720

### CODE REVIEW

File: src/Controls/tests/TestCases/Issues/Issue18720DatePicker.xaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases/Issues/Issue18720DatePicker.xaml:1]: fix-18720-datepicker

File: src/Controls/tests/TestCases/Issues/Issue18720Editor.xaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases/Issues/Issue18720Editor.xaml:1]: fix-18720-editor

File: src/Controls/tests/TestCases/Issues/Issue18720TimePicker.xaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases/Issues/Issue18720TimePicker.xaml:1]: fix-18720-timepicker

**Comment by mattleibow** [comment]
I updated this PR to not need a flag but rather use a know type of drawable. If the drawable used in the background is a Maui*Drawable, then we know that we added it and can remove it. Otherwise, we don't do anything since the only alternative is that it was the user.

There is the case where the cross-platform view has a background set, then we keep existing behavior and always override the background with the one set in the view. This means that if you have a custom handler that is going things AND set the background in XAML, the XAML will still win and override your custom background.

However, if you are trying to replace the background, make sure you are replacing the mapper:

```cs
builder.ConfigureMauiHandlers(handlers =>
{
#if ANDROID
    EntryHandler.Mapper.ReplaceMapping<IEntry, IEntryHandler>("Background", (handler, view) =>
    {
        var platformView = handler.PlatformView;

        // Android will reset the padding when setting a Background drawable, so we need to reapply the padding after
        var padLeft = platformView.PaddingLeft;
        var padTop = platformView.PaddingTop;
        var padRight = platformView.PaddingRight;
        var padBottom = platformView.PaddingBottom;

        // Set the custom background
        var gd = new Android.Graphics.Drawables.GradientDrawable();
        gd.SetCornerRadius(3);
        gd.SetStroke(1, Android.Graphics.Color.Red);
        gd.SetColor(Android.Graphics.Color.Pink);
        platformView.Background = gd;

        // Apply previous padding
        platformView.SetPadding(padLeft, padTop, padRight, padBottom);
    });
#endif
});
```

**Review by mattleibow — APPROVED** [review]

**Pull request merged**